pkg/peertls/extensions.go

```diff
@@ -235,46 +234,40 @@ func (e ExtensionHandlers) VerifyFunc() PeerCertVerificationFunc {
 func (r Revocation) Verify(signingCert *x509.Certificate) error {
 	pubKey, ok := signingCert.PublicKey.(crypto.PublicKey)
 	if !ok {
-		return ErrUnsupportedKey.New("%T", signingCert.PublicKey)
-	}
-
-	data, err := r.TBSBytes()
-	if err != nil {
-		return err
+		return pkcrypto.ErrUnsupportedKey.New("%T", signingCert.PublicKey)
 	}
 
-	if err := VerifySignature(r.Signature, data, pubKey); err != nil {
+	data := r.TBSBytes()
+	if err := pkcrypto.VerifySignature(r.Signature, data, pubKey); err != nil {
 		return err
 	}
 	return nil
 }
 
 // TBSBytes (ToBeSigned) returns the hash of the revoked certificate hash and
 // the timestamp (i.e. hash(hash(cert bytes) + timestamp)).
-func (r *Revocation) TBSBytes() ([]byte, error) {
-	toHash := new(bytes.Buffer)
+func (r *Revocation) TBSBytes() []byte {
+	var toHash bytes.Buffer
 	_, err := toHash.Write(r.CertHash)
 	if err != nil {
-		return nil, ErrExtension.Wrap(err)
+		// bytes.Buffer#Write() is documented as never returning an error
+		panic(err)
 	}
 
 	// NB: append timestamp to revoked cert bytes
 	binary.PutVarint(toHash.Bytes(), r.Timestamp)
 
-	return SHA256Hash(toHash.Bytes())
+	return pkcrypto.SHA256Hash(toHash.Bytes())
 }
 
 // Sign generates a signature using the passed key and attaches it to the revocation.
 func (r *Revocation) Sign(key crypto.PrivateKey) error {
-	data, err := r.TBSBytes()
-	if err != nil {
-		return err
-	}
-
-	r.Signature, err = signHashOf(key, data)
+	data := r.TBSBytes()
+	sig, err := pkcrypto.SignHashOf(key, data)
 	if err != nil {
 		return err
 	}
+	r.Signature = sig
```

Comment:
is it bad form to assign directly like it was doing before? (`r.Signature, err := ...`)

Reply:
One is more defensive about the changing of structures. When something returns invalid data and an error, then this can prevent some bugs. But assuming well behaved data to be returned then it shouldn't matter. I generally prefer the shorter form.

Reply:
I do tend to prefer a more defensive approach. Want me to change it to the shorter?

Reply:
I have no strong feeling about this.